addons/misc/functions/fnc_onDeathZeus.sqf

```diff
@@ -0,0 +1,184 @@
+
+*///////////////////////////////////////////////
+
+
+
+params [["_pos",[0,0,0],[[]],3], ["_unit",objNull,[objNull]]];
+
+if(isNull _unit) exitWith { hint "Must be placed on a unit or object"; };
+
+private _onConfirm =
+{
+	params ["_dialogResult","_in"];
+
+	private _expectedParams = [
+		"_hint_killer",
+		"_hint_all",
+		"_hint_zeus",
+		"_deleteRemains",
+		"_explosion",
+		"_customCode",
+		"_codeTarget"
+	];
+
+	if (isClass(configFile >> "CfgPatches" >> "crowsEW_main")) then {
+		_expectedParams insert [5, ["_emp", "_sound_killer", "_sound_all"]];
+	};
+
+	_dialogResult params _expectedParams;
+	_in params ["_pos", ["_unit",objNull,[objNull]]];
+
+	if(isNull _unit) exitWith { hint "Something has gone wrong!"; diag_log "CrowsZA-onDeathZeus: unit became invalid after selection";};
+
+	if(
+		_hint_killer isEqualTo "" &&
+		{_hint_all isEqualTo "" &&
+		{_hint_zeus isEqualTo "" &&
+		{!_deleteRemains &&
+		{_explosion == 0 &&
+		{_customCode isEqualTo "" &&
+		{isNil "_emp" || {!_emp}} &&
+		{isNil "_sound_killer" || {_sound_killer isEqualTo ""}} &&
+		{isNil "_sound_all" || {_sound_all isEqualTo ""}}}}}}}
+	) exitWith { hint "Nothing set - no action taken!" };
+
+
+	private _unitList = missionNamespace getVariable ["crowza_onKilledEHUnits", []];
+	_unitList pushBackUnique _unit;
+	missionNamespace setVariable ["crowza_onKilledEHUnits", _unitList, true];
+
+	_unit setVariable [QGVAR(onDeath_hintKiller), _hint_killer];
+	_unit setVariable [QGVAR(onDeath_hintAll), _hint_all];
+	_unit setVariable [QGVAR(onDeath_hintZeus), _hint_zeus];
+	_unit setVariable [QGVAR(onDeath_deleteRemains), _deleteRemains];
+	_unit setVariable [QGVAR(onDeath_explosion), _explosion];
+	_unit setVariable [QGVAR(onDeath_emp), _emp];
+	_unit setVariable [QGVAR(onDeath_soundKiller), _sound_killer];
+	_unit setVariable [QGVAR(onDeath_soundAll), _sound_all];
+	_unit setVariable [QGVAR(onDeath_customCode), _customCode];
+	_unit setVariable [QGVAR(onDeath_codeTarget), _codeTarget];
+
+	// If the EH is being overwritten (i.e. the module has been applied already), remove the old one first
+	private _ehIndex = _unit getVariable ["crowza_onKilledEHIndex", nil];
+	if(!isNil "_ehIndex") then {
+		_unit removeMPEventHandler ["MPKilled", _ehIndex];
+	};
+
+	_ehIndex = _unit addMPEventHandler ["MPKilled", {
+		params ["_unit", "_killer", "_instigator", "_useEffects"];
+
+		private _hint_killer = _unit getVariable [QGVAR(onDeath_hintKiller), ""];
+		if(_hint_killer isNotEqualTo "" and player isEqualTo _killer) then {
+			hint _hint_killer;
+		};
+
+		private _hint_all = _unit getVariable [QGVAR(onDeath_hintAll), ""];
+		if(_hint_all isNotEqualTo "") then {
+			hint _hint_all;
+		};
+
+		private _hint_zeus = _unit getVariable [QGVAR(onDeath_hintZeus), ""];
+		if(_hint_zeus isNotEqualTo "" and !isNull (getAssignedCuratorLogic player)) then {
+			hint _hint_zeus;
+		};
+
+		// TODO: worth investigating what happens if crowsEW is present ONLY on this client
```

Comment:
You can disregard that. If EW is only present on this client, then most of EW will be broken anyway. EW is set like CBA, ZEN and others, to require everyone to load it.